File: frontend/src/app/core/models/watches/watch-full-info-response.dto.ts
import { WatchCondition } from './enums/watch-condition';
import { WatchGender } from './enums/watch-gender';
import { WatchMovementType } from './enums/watch-movement-type';
import { WatchStatus } from './enums/watch-status';
import { WatchType } from './enums/watch-type';
import { BranchShortResponseDTO } from '../branch/branch-short-response.dto';
import { WatchPhotoShortResponseDTO } from './photos/watch-photo-short-response.dto';

export interface WatchFullInfoResponseDTO {
    id: string;
    manufacturer: string;
    model: string;
    movement: string;
    referenceNumber: string;
    serialNumber: string;
    description: string;
    yearOfProduction: number;
    pricePerDay: number;
    condition: WatchCondition;
    gender: WatchGender;
    movementType: WatchMovementType;
    status: WatchStatus;
    watchType: WatchType;
    branch: BranchShortResponseDTO;
    photos: WatchPhotoShortResponseDTO[]
}

File: backend/src/main/java/com/yourwatchrental/watchrental/watch/dto/response/WatchFullInfoResponseDTO.java
package com.yourwatchrental.watchrental.watch.dto.response;

import com.yourwatchrental.watchrental.branch.dto.BranchShortResponseDTO;
import com.yourwatchrental.watchrental.watch.enums.*;
import com.yourwatchrental.watchrental.watch.watchphoto.dto.WatchPhotoShortResponseDTO;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record WatchFullInfoResponseDTO(
        UUID id,
        String manufacturer,
        String model,
        String referenceNumber,
        String serialNumber,
        String movement,
        String description,
        Integer yearOfProduction,
        BigDecimal pricePerDay,
        Condition condition,
        Gender gender,
        MovementType movementType,
        Status status,
        WatchType watchType,
        BranchShortResponseDTO branch,
        List<WatchPhotoShortResponseDTO> photos
) {
}
